CRM/Dedupe/BAO/RuleGroup.php

```diff
@@ -196,13 +196,13 @@ public function fillTable() {
     if ($this->params && !$this->noRules) {
       $tempTableQuery = "CREATE TEMPORARY TABLE dedupe (id1 int, weight int, UNIQUE UI_id1 (id1)) ENGINE=InnoDB";
       $insertClause = "INSERT INTO dedupe (id1, weight)";
-      $groupByClause = "GROUP BY id1";
+      $groupByClause = "GROUP BY id1, weight";
```

Comment:
the other option could be max(weight) i suppose

Reply:
I would say MAX(weight) would be safe.

Reply:
I'm finding it harder to get comfortable with this - I'd probably prefer the rule group changes in a separate PR so I can get my head around them without blocking the ohters